### PR TITLE
Modernize gnsq for nsq 1.0.0 and python 3.7

### PR DESCRIPTION
**Pull request by wtolson**

### CODE REVIEW

File: docs/upgrading.rst [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+:class:`messages <gnsq.Message>` has a
+:func:`message.enable_async() <gnsq.Message.enable_async>`
+method that may be used to indicate that a message will be handled
+asynchronous.
+
+Max concurrency
+```````````````
+
+The `max_concurrency` parameter has been removed from :class:`gnsq.Consumer`.
+If you with to replicate this behavior, you should use the
```
Comment (fmoor) [inline docs/upgrading.rst]: > If you **with** to replicate

probably `wish`?

File: docs/upgrading.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        pass
+
+Deprecated Nsqd
+~~~~~~~~~~~~~~~
+
+The :class:`gnsq.Nsqd` client has been split into two classes, corresponding
+to the tcp and http APIs. The new classes are :class:`gnsq.NsqdTCPClient` and
+:class:`gnsq.NsqdHTTPClient` respectively.
+
+The methods `publish_tcp`, `publich_http`, `multipublish_tcp`, and
```
Comment (fmoor) [inline docs/upgrading.rst]: should this be `publish_http`?

```suggestion
The methods `publish_tcp`, `publish_http`, `multipublish_tcp`, and
```

File: gnsq/contrib/batch.py [hunk 3]
```diff
@@ -14,12 +14,11 @@
 class BatchHandler(object):
     """Batch message handler for gnsq.
 
-    The batch handler assumes the reader is in async mode and the inflight is
-    larger then the batch size.
+    The batch handler assumes the max inflight is larger then the batch size.
```
Comment (fmoor) [inline gnsq/contrib/batch.py]:
```suggestion
    The batch handler assumes the max inflight is larger than the batch size.
```

Comment (wtolson) [reply gnsq/contrib/batch.py]: Good catch, updated.

**Pull request merged**